### PR TITLE
[KOGITO-1861] - Make possible to turn the Maven downloading process output off during s2i builds

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Jira issue : https://issues.redhat.com/browse/KOGITO-1861

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: deploy/crds/app.kiegroup.org_kogitoapps_crd.yaml [hunk 1]
```diff
@@ -127,9 +127,13 @@ spec:
                                   optional for env vars'
                                 type: string
                               divisor:
+                                anyOf:
```
Comment (spolti) [inline deploy/crds/app.kiegroup.org_kogitoapps_crd.yaml:130]: other changes looks good, but, why this was changed?
is you using the correct operator-sdk version?

Comment (vaibhavjainwiz) [reply deploy/crds/app.kiegroup.org_kogitoapps_crd.yaml:130]: I am using operator-sdk version: "v0.17.0". Is it correct?

Comment (spolti) [reply deploy/crds/app.kiegroup.org_kogitoapps_crd.yaml:130]: no, we still on 0.15.1;

**Comment by sutaakar** [comment]
Looks good, will approve once rebuilt with proper operator-sdk version

File: deploy/crds/app.kiegroup.org_kogitoapps_crd.yaml (hunk 1 above)
Comment (vaibhavjainwiz) [reply deploy/crds/app.kiegroup.org_kogitoapps_crd.yaml]: changed operator-sdk version 0.15.1 and regenerate code

**Comment by vaibhavjainwiz** [comment]
> Looks good, will approve once rebuilt with proper operator-sdk version

@sutaakar I had changed operator-sdk version to 0.15.1 and regenerated the code

**Comment by ricardozanini** [comment]
@spolti I believe we can merge this one :)

**Comment by spolti** [comment]
@radtriste can you please take a look?

File: cmd/kogito/command/deploy/deploy_service.go [hunk 2]
```diff
@@ -47,24 +47,25 @@ var (
...
+	imageRuntime        string
+	native              bool
+	buildLimits         []string
+	buildRequests       []string
+	imageVersion        string
+	mavenMirrorURL      string
+	enableIstio         bool
+	enablePersistence   bool
+	enableEvents        bool
+	mavenDownloadOutput bool
```
Comment (radtriste) [inline cmd/kogito/command/deploy/deploy_service.go:68]: Could we have something like `enableMavenDownloadOutput` for the option as it is a boolean value ?

File: deploy/crds/app.kiegroup.org_v1alpha1_kogitoapp_cr.yaml [hunk 3]
```diff
@@ -8,4 +8,6 @@ spec:
       uri: https://github.com/kiegroup/kogito-examples
       contextDir: process-quarkus-example
     # set your maven nexus repository
-    #mavenMirrorURL: http://nexus3-nexus.apps-crc.testing/repository/maven-public/
\ No newline at end of file
+    #mavenMirrorURL: http://nexus3-nexus.apps-crc.testing/repository/maven-public/
+    # enable to print maven transfer logs in console
+    #maven-download-output: true
```
Comment (radtriste) [inline deploy/crds/app.kiegroup.org_v1alpha1_kogitoapp_cr.yaml:13]: could we have it like `mavenMirrorURL`, I mean it should be `mavenDownloadOutput` field in all examples

File: pkg/apis/app/v1alpha1/kogitoapp_types.go [hunk 4]
```diff
@@ -131,6 +131,8 @@ type KogitoAppBuildObject struct {
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
 	// Internal Maven Mirror to be used during source-to-image builds to considerably increase build speed
 	MavenMirrorURL string `json:"mavenMirrorURL,omitempty"`
+	// Suppress the logs of downloading/uploading maven dependencies
```
Comment (radtriste) [inline pkg/apis/app/v1alpha1/kogitoapp_types.go:134]:
```suggestion
	// Show the logs of downloading/uploading maven dependencies
```

**Comment by radtriste** [comment]
depends on https://github.com/kiegroup/kogito-images/pull/165

File: deploy/crds/app.kiegroup.org_v1alpha1_kogitoapp_cr.yaml (hunk 3 above)
Comment (spolti) [reply deploy/crds/app.kiegroup.org_v1alpha1_kogitoapp_cr.yaml:13]: +1

File: pkg/apis/app/v1alpha1/kogitoapp_types.go (hunk 4 above)
Comment (spolti) [reply pkg/apis/app/v1alpha1/kogitoapp_types.go:134]: wouldn't be better something like "switch to print or not dependencies download/upload status"

File: cmd/kogito/command/deploy/deploy_service.go (hunk 2 above)
Comment (vaibhavjainwiz) [reply cmd/kogito/command/deploy/deploy_service.go]: changed name to enableMavenDownloadOutput

File: deploy/crds/app.kiegroup.org_v1alpha1_kogitoapp_cr.yaml (hunk 3 above)
Comment (vaibhavjainwiz) [reply deploy/crds/app.kiegroup.org_v1alpha1_kogitoapp_cr.yaml]: changed name to enableMavenDownloadOutput

File: pkg/apis/app/v1alpha1/kogitoapp_types.go (hunk 4 above)
Comment (vaibhavjainwiz) [reply pkg/apis/app/v1alpha1/kogitoapp_types.go]: set descriptio to `If set to true will print the logs for downloading/uploading of maven dependencies. Defaults to false`

File: pkg/controller/kogitoapp/resource/build_config_s2i.go [hunk 5]
```diff
@@ -36,6 +36,7 @@ const (
 	buildS2IlimitCPUEnvVarKey    = "LIMIT_CPU"
 	buildS2IlimitMemoryEnvVarKey = "LIMIT_MEMORY"
 	mavenMirrorURLEnvVar         = "MAVEN_MIRROR_URL"
+	mavenDownloadOutput          = "MAVEN_DOWNLOAD_OUTPUT"
```
Comment (radtriste) [inline pkg/controller/kogitoapp/resource/build_config_s2i.go:39]: change to `mavenDownloadOutputEnvVar`

Comment (vaibhavjainwiz) [reply pkg/controller/kogitoapp/resource/build_config_s2i.go]: chnaged name to mavenDownloadOutputEnvVar

File: cmd/kogito/command/deploy/deploy_service.go [hunk 6]
```diff
@@ -175,7 +175,7 @@ func (i *deployCommand) InitHook() {
 	i.command.Flags().StringVar(&i.flags.projectGroupId, "project-group-id", "", "Indicates the unique identifier of the organization or group that created the project, e.g: 'com.company'")
 	i.command.Flags().StringVar(&i.flags.projectArtifactId, "project-artifact-id", "", "Indicates the unique base name of the primary artifact being generated, e.g: 'project'")
 	i.command.Flags().StringVar(&i.flags.projectVersion, "project-version", "", "Indicates the version of the artifact generated by the project, e.g: '1.0-SNAPSHOT'")
-  i.command.Flags().BoolVar(&i.flags.enableMavenDownloadOutput, "enable-maven-download-output", false, "If set to true will print the logs for downloading/uploading of maven dependencies. Defaults to false")
+	i.command.Flags().BoolVar(&i.flags.enableMavenDownloadOutput, "enable-maven-download-output", false, "If set to true will print the logs for downloading/uploading of maven dependencies. Defaults to false")
```
Comment (spolti) [inline cmd/kogito/command/deploy/deploy_service.go:178]: why not only maven-output?
and, maybe use BoolVarP to set a short-hand option for this as well?

Comment (vaibhavjainwiz) [reply cmd/kogito/command/deploy/deploy_service.go]: I had changes name to **maven-output**. I didn't find any single character which we can use here as short hand. I check "o" but its already taken. Could you please suggest which we short hand character we can use here?

Comment (vaibhavjainwiz) [reply cmd/kogito/command/deploy/deploy_service.go]: using "m" as short hand char

**Comment by spolti** [comment]
let's wait for CI before merge.

**Pull request merged**